### DISCUSSION TITLE
The meaning of --scale_min_memory option for preprocessing

### DISCUSSION THREAD

**Discussion by geowk** [CoBrALab/RABIES#357] [General]
If I use --local_threads 8 and --scale_min_memory 2 for preprocessing, what does the number 2 indicate (the default value is 1)? 

Is it placing some type of cap on memory usage if the memory exceeds a threshold limit while running 8 threads concurrently?

Thanks.

-g

**Comment by Gab-D-G** [comment]
Certain nipype nodes have a minimum memory allocation for running the node, and this prevents running too many operations simultaneously if not enough memory is available. --scale_min_memory is a multiplier that will increase or decrease this memory requirement (1.0 means the same as default, 0.5 is half from default meaning operations need less memory, and 2.0 would be double the memory requirement). So if you increase the multiplier it could mitigate memory errors if that is the issue.

**Comment by geowk** [comment]
Thanks for the response

> So if you increase the multiplier it could mitigate memory errors if that is the issue.

Errors due to the node's cap on memory or computer's ram?

In this scenario, does this mean that node has reached its requirement limit despite more memory available on the computer system? 

Let's say the node minimum memory requirement is 10 GB? If I use --scale_min_memory 2,does  this mean 20 GB is available for processing instead of 10 GB?

-g

**Reply by Gab-D-G** [reply]
Let's say by default a given node allocates 1Gb to do its job (a rough estimate of how much memory would be required for it, provided a standard image size), then --scale_min_memory 2.0 will increase this allocation to 2Gb, meaning that the total memory cap (let's say 10Gb of RAM on your PC) will be reached more quickly, and so it will limit how many processes will be allowed to run simultaneously. The cap on available computer RAM is never changed, it is computed automatically. Does that sound clearer?

Best,
Gabe

**Comment by geowk** [comment]
Thanks, Gabe.  Yes, I understand now. This helps a lot.

Thanks again.

-g
